components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/util/CommonUtils.java

```diff
@@ -0,0 +1,266 @@
+
+public class CommonUtils {
+    private static final String ELEMENT_ID = "id";
+    private static final String PREVIOUS_ELEMENT_PATH = "previousSibling";
+    private static final String CURRENT_ELEMENT_PATH = "currentElement";
+    private static final String NEXT_ELEMENT_PATH = "nextSibling";
+    private static final String DISPLAY_NAME = "displayName";
+    // two different usage of id constant
+    private static final String ID = "id";
+    private static final String TEXT = "label";
+    private static final String CHILDREN = "children";
+
+    /**
+     * This method will return list of rxt bean objects with taxonomy meta data
+     *
+     * @return rxt bean objects
+     * @throws UserStoreException
+     * @throws RegistryException
+     */
+    public static List<RXTBean> getRxtTaxonomies() throws UserStoreException, RegistryException {
+
+        List<RXTBean> rxtBeenList = new ArrayList<>();
+        int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        String adminName = ServiceHolder.getRealmService().getTenantUserRealm(tenantId).getRealmConfiguration()
+                .getAdminUserName();
+        Registry registry = ServiceHolder.getRegistryService().getGovernanceUserRegistry(adminName, tenantId);
+        RealmService realmService = registry.getRegistryContext().getRealmService();
+
+        String configurationPath = RegistryConstants.GOVERNANCE_COMPONENT_PATH + "/types/";
+        if (realmService.getTenantUserRealm(realmService.getTenantManager().getTenantId(tenantDomain))
+                .getAuthorizationManager().isUserAuthorized(adminName, configurationPath, ActionConstants.GET)) {
+
+            GovernanceUtils.loadGovernanceArtifacts((UserRegistry) registry);
+            List<GovernanceArtifactConfiguration> configurations = GovernanceUtils.
+                    findGovernanceArtifactConfigurations(registry);
+            for (GovernanceArtifactConfiguration governanceArtifactConfiguration : configurations) {
+                RXTBean rxtBean = new RXTBean();
+                rxtBean.setTaxonomy(governanceArtifactConfiguration.getTaxonomy());
+                rxtBean.setRxtName(governanceArtifactConfiguration.getKey());
+                rxtBeenList.add(rxtBean);
+            }
+
+        }
+        return rxtBeenList;
+
+    }
+
+    /**
+     * This method will process the rest api request and return json array
+     *
+     * @param query        user query request
+     * @param updatedQuery updated query
+     * @param startNode    start node
+     * @param endNode      end node
+     * @param nodeList     xml node list
+     * @return json array
+     * @throws TaxonomyException
+     * @throws JSONException
+     */
+    public static JSONArray toJson(String query, String updatedQuery, int startNode, int endNode, NodeList nodeList)
```

Comment:
This is a very complex method. Methods should describe only a single entity. If possible please reduce the complexity by breaking down into different methods and reusing them.